### PR TITLE
OSASINFRA-3229: Remove support for Kuryr

### PR DESCRIPTION
**Pull request by dulek**
In 4.15 Kuryr is no longer a supported NetworkType, following its deprecation in 4.12. This commit removes mentions of Kuryr from the documentation and code, but also adds validation to prevent installations from being executed when `networkType` is set to `Kuryr`.

### CODE REVIEW

**Comment by dulek** [comment]
/test e2e-openstack-kuryr

Just to see if it'll fail as expected.

File: pkg/types/validation/installconfig.go [hunk 1]
```diff
@@ -329,6 +329,10 @@ func validateNetworking(n *types.Networking, singleNodeOpenShift bool, fldPath *
 		allErrs = append(allErrs, field.Required(fldPath.Child("networkType"), "network provider type required"))
 	}
 
+	if n.NetworkType == string(operv1.NetworkTypeKuryr) {
```
Comment (mandre) [inline pkg/types/validation/installconfig.go:332]: Shouldn't we have a list of valid network types coming from the API, and error when the user provides an unknown network type? If so, we should start by removing the Kuryr network type from the API, then simply bump the installer to the new API.

File: upi/openstack/inventory.yaml [hunk 2]
```diff
@@ -19,8 +19,8 @@ all:
       # Subnet pool prefix length
       host_prefix: '23'
       # Name of the SDN.
-      # Possible values are OpenshiftSDN or Kuryr.
-      os_networking_type: 'OpenshiftSDN'
+      # Possible values are OVNKubernetes or OpenShiftSDN
+      os_networking_type: 'OVNKubernetes'
```
Comment (mandre) [inline upi/openstack/inventory.yaml:23]: While I agree that OVNKubernetes should now be the default, this is a functional, unrelated, change and possibly belongs to a different PR.

File: pkg/types/validation/installconfig.go (hunk 1 above)
Comment (dulek) [reply pkg/types/validation/installconfig.go:332]: There is no such validation in the installer. I've just started installation with `networking.type="Foobar" it will only fail on CNO level. With this patch I want to clean up Kuryr bits, I don't want to introduce new validations that are not related to Kuryr.

File: upi/openstack/inventory.yaml (hunk 2 above)
Comment (dulek) [reply upi/openstack/inventory.yaml:23]: Agreed, fixing that now! I believe this thinking can also be applied to your other comment - the fact that no networkType validation is in place is unrelated to Kuryr.

Comment (mandre) [reply upi/openstack/inventory.yaml]: Yes, I also think that adding validation for the network type should be its own PR.

File: docs/user/openstack/install_upi.md [hunk 3]
```diff
@@ -418,11 +407,11 @@ $ python -c '
 import yaml
 path = "install-config.yaml"
 data = yaml.safe_load(open(path))
-data["networking"]["networkType"] = "Kuryr"
+data["networking"]["networkType"] = "OpenShiftSDN"
 open(path, "w").write(yaml.dump(data, default_flow_style=False))'
 ```
 
-Also set `os_networking_type` to `Kuryr` in `inventory.yaml`.
+Also set `os_networking_type` to `OpenShiftSDN` in `inventory.yaml`.
```
Comment (MaysaMacedo) [inline docs/user/openstack/install_upi.md:414]: `os_networking_type` was only used to enable creation of openstack resources needed by Kuryr, it[ loos like](https://github.com/search?q=repo%3Aopenshift%2Finstaller%20os_networking_type&type=code) we can remove it from the inventory.

File: upi/openstack/inventory.yaml [hunk 4]
```diff
@@ -19,8 +19,8 @@ all:
       # Subnet pool prefix length
       host_prefix: '23'
       # Name of the SDN.
-      # Possible values are OpenshiftSDN or Kuryr.
-      os_networking_type: 'OpenshiftSDN'
+      # Possible values are OVNKubernetes or OpenShiftSDN
+      os_networking_type: 'OpenShiftSDN'
```
Comment (MaysaMacedo) [inline upi/openstack/inventory.yaml:23]: All the variables defined from line 15-23 can now be removed.

**Comment by MaysaMacedo** [comment]
The [destroy script](https://github.com/openshift/installer/blob/master/pkg/destroy/openstack/openstack.go) for IPI could be updated as well. For example the `deleteSubnetPools` function could be removed, there might be other functions that can be removed.

**Comment by dulek** [comment]
> The [destroy script](https://github.com/openshift/installer/blob/master/pkg/destroy/openstack/openstack.go) for IPI could be updated as well. For example the `deleteSubnetPools` function could be removed, there might be other functions that can be removed.

I've removed the subnet pools deletion. I see that we could simplify LBs deletion as all supported OpenStack versions support tagging in Octavia. But besides that everything else seems to be required.

**Comment by MaysaMacedo** [comment]
looks like dependency on subnetpools should be removed from the vendor folder.
Maybe `go mod vendor` can help with that

**Comment by dulek** [comment]
> looks like dependency on subnetpools should be removed from the vendor folder. Maybe `go mod vendor` can help with that

Yeah, it was not working yesterday as it couldn't download one of the packages. All good now, thank you!

File: pkg/types/validation/installconfig.go (hunk 1 above)
Comment (mandre) [reply pkg/types/validation/installconfig.go:333]: Do we really need this validation then, if the installer happily accepts any network type?
We don't want to depend on `NetworkTypeKuryr` as we're removing it from the API.

Comment (dulek) [reply pkg/types/validation/installconfig.go:333]: My point with this is to give users feedback on the installer level that Kuryr is not supported anymore. Otherwise installer will happily accept `networkType=Kuryr` and user will only know that this doesn't work once network-operator gets up and reports (or not) an error. How about we change that to a hardcoded string to anticipate the API removal?

Comment (mandre) [reply pkg/types/validation/installconfig.go:333]: That works for me. At some point, it would also be good to add a more generic validation to the installer, so that users with fat fingers don't spend hours debugging their failed deployments.

Comment (dulek) [reply pkg/types/validation/installconfig.go]: Done, thanks!

**Review by mandre — COMMENTED** [review]
/lgtm
/approve

@patrickdillon whenever you get a chance to review, some bits (vendoring and validations) are outside of our control and we can't approve.
/assign @patrickdillon

**Review by r4f4 — COMMENTED** [review]
/approve
